testsuite/pytests/sli2py_synapses/test_inh_stdp_synapse.py

```diff
@@ -0,0 +1,221 @@
+
+Synopsis: (test_inh_stdp_synapse) run
+
+Description:
+  A parrot_neuron that repeats the spikes from a poisson generator is
+  connected to an iaf_psc_alpha that is driven by inh. and exc. poisson input.
+  The synapse is an stdp_synapse. After the simulation, we go through the pre-
+  and postsyn. spike-trains spike by spike and try to reproduce the STDP
+  results. The final weight obtained after simulation is compared to the final
+  weight obtained from the test.
+
+Test ported from SLI unittest.
+
+Author: Weidel Mar 2016
+Based on: Kunkel, Nov 2010
+"""
+
+import nest
+import numpy as np
+import pytest
+
+
+def test_inh_stdp_synapse():
+    """
+    Test STDP synapse by reproducing weight changes from spike trains.
+    """
+    nest.ResetKernel()
+
+    resolution = 0.1
+    nest.set(resolution=resolution)
+
+    # Input parameters
+    K_exc = 8000.0  # number of exc. inputs
+    K_inh = 2000.0  # number of inh. inputs
+    nu = 10.0  # equil. firing rate
+    nu_x = 1.7  # external rate
+    w_exc = 45.0  # strength of exc. connections
+    w_neuron = -45.0  # strength of connections between neuron and parrot
+    w_inh = w_exc * -5.0  # strength of inh. connections
+    delay = 1.0  # synaptic transmission delay
+
+    axonal_delay = 0.0
+    backpr_delay = delay - axonal_delay
+
+    # STDP parameters
+    alpha = 1.1
+    lambda_val = 0.01
+    tau_plus = 20.0
+    tau_minus = 30.0
+    mu_plus = 1.0  # multiplicative
+    mu_minus = 1.0  # multiplicative
+    w_max = w_neuron * 2.0
+
+    # Create poisson generators, neurons and spike recorder
+    pg_exc = nest.Create("poisson_generator", params={"rate": K_exc * (nu + nu_x)})
+    pg_inh = nest.Create("poisson_generator", params={"rate": K_inh * nu})
+    pg_pre = nest.Create("poisson_generator", params={"rate": nu})
+
+    parrot = nest.Create("parrot_neuron")
+    neuron = nest.Create("iaf_psc_alpha", params={"tau_minus": tau_minus})
+
+    sr_pre = nest.Create("spike_recorder")
+    sr_post = nest.Create("spike_recorder")
+
+    # Set defaults for stdp_synapse
+    # Note: "lambda" is a Python keyword, but can be used as a dict key
+    nest.SetDefaults(
+        "stdp_synapse",
+        {
+            "weight": w_neuron,
+            "alpha": alpha,
+            "lambda": lambda_val,  # noqa: A001
+            "tau_plus": tau_plus,
+            "mu_plus": mu_plus,
+            "mu_minus": mu_minus,
+            "Wmax": w_max,
+        },
+    )
+
+    # Connect
+    nest.Connect(pg_exc, neuron, syn_spec={"weight": w_exc, "delay": delay})
+    nest.Connect(pg_inh, neuron, syn_spec={"weight": w_inh, "delay": delay})
+    nest.Connect(pg_pre, parrot, syn_spec={"weight": w_exc, "delay": delay})
+
+    nest.Connect(parrot, neuron, syn_spec={"synapse_model": "stdp_synapse", "weight": w_neuron, "delay": delay})
+
+    nest.Connect(parrot, sr_pre)
+    nest.Connect(neuron, sr_post)
+
+    # Simulate and get data
+    nest.Simulate(10000.0)
```

Comment:
Maybe for consistency with other tests:
```suggestion
    nest.Simulate(T)
```
and `T = 10000.0` at the top.